alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -51,8 +52,20 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
 
     suspend fun initDocLevelQueryIndex(): Boolean {
         if (!docLevelQueryIndexExists()) {
+            // Since we changed queryIndex to be alias now, for backwards compatibility, we have to delete index with same name
+            // as our alias, to avoid name clash.
+            if (clusterService.state().metadata.hasIndex(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)) {
+                val acknowledgedResponse: AcknowledgedResponse = client.suspendUntil {
+                    admin().indices().delete(DeleteIndexRequest(ScheduledJob.DOC_LEVEL_QUERIES_INDEX))
+                }
+                if (!acknowledgedResponse.isAcknowledged) {
+                    val errorMessage = "Deletion of old queryIndex [${ScheduledJob.DOC_LEVEL_QUERIES_INDEX}] index is not acknowledged!"
+                    log.error(errorMessage)
+                    throw AlertingException.wrap(OpenSearchStatusException(errorMessage, RestStatus.INTERNAL_SERVER_ERROR))
+                }
+            }
```

Comment:
Just to confirm, even when it gets deleted here, we don't care for the old data since we will recreate it anyways, right?

Reply:
Yes, it is checked if it exist and recreated if not, on every monitor update/execute API call and every monitor execution by job scheduler